### DISCUSSION TITLE
How to retrieve user specific jobs from queue?

### DISCUSSION THREAD

**Discussion by aditodkar** [OptimalBits/bull#2243] [General]
I have created single queue in which multiple logged in users can add job to it. Now on dashboard I want to show all the jobs which are completed/delayed/failed for that particular user. I using uuidv4 for job id as well as user id is of type uuidv4. Now how to retrieve jobs of particular logged in user? For this should I create job id and user id mapping in database and then query all the jobs using [getJob ](https://github.com/OptimalBits/bull/blob/develop/REFERENCE.md#queuegetjob) or is there any better way to do this? Please suggest. Thanks

**Comment by manast** [comment]
For this kind of meta functionality, you have two options, either offload data to a separate database with query capabilities or implement custom Redis code for adding indexes and searches on the data. I am aware alternative 2 is not particularly easy to achieve currently...

**Reply by aditodkar** [reply]
I have used first approach as mentioned by you. Thanks a lot :)

**Reply by aditodkar** [reply]
@manast  Is there any way to get job data of repeatable job? I am trying to query repeatable job specific to a user. Something like this:
```
await queue.add('remove', { example: 'data' }, { jobId: 'findMe', repeat: { every: 1000 } })

// ... then later ...
const repeatableJobs = await queue.getRepeatableJobs()
const foundJob = repeatableJobs.find(job => job.id === 'findMe')
```
or should I store job data in database as I am already storing user and job id mapping for regular jobs in database but in case of regular job I did not store job data inside database because I was able to get job Data using this: https://github.com/OptimalBits/bull/blob/develop/REFERENCE.md#queuegetjob

Any suggestions? Thanks in advance :)

**Reply by manast** [reply]
Unfortunately, we do not have better "getters" for the repeatable jobs atm. So if you do not have a lot of them, your approach above will work well enough, is that your case?

**Reply by aditodkar** [reply]
I have an application in which each user can schedule some tasks. And on dashboard page I want to show 3 things:
1) Failed jobs
2) Completed jobs
3) Delayed or scheduled jobs
In all the 3 cases I want to show job data as well not just count specific to that user.
